src/CommunityToolkit.Maui/Behaviors/PlatformBehaviors/Touch/GestureManager.cs

```diff
@@ -0,0 +1,789 @@
+		if (status != TouchStatus.Started || canExecuteAction)
+		{
+			if (!canExecuteAction)
+			{
+				status = TouchStatus.Canceled;
+			}
+
+			var state = status == TouchStatus.Started
+				? TouchState.Pressed
+				: TouchState.Normal;
+
+			if (status == TouchStatus.Started)
+			{
+				animationProgress = 0;
+				animationState = state;
+			}
+
+			var isToggled = sender.IsToggled;
+			if (isToggled.HasValue)
+			{
+				if (status != TouchStatus.Started)
+				{
+					durationMultiplier = (animationState == TouchState.Pressed && !isToggled.Value) ||
+						(animationState == TouchState.Normal && isToggled.Value)
+						? 1 - animationProgress
+						: animationProgress;
+
+					UpdateStatusAndState(sender, status, state);
+
+					if (status == TouchStatus.Canceled)
+					{
+						sender.ForceUpdateState(false);
+						return;
+					}
+
+					OnTapped(sender);
+					sender.IsToggled = !isToggled;
+					return;
+				}
+
+				state = isToggled.Value
+					? TouchState.Normal
+					: TouchState.Pressed;
+			}
+
+			UpdateStatusAndState(sender, status, state);
+		}
+
+		if (status == TouchStatus.Completed)
+		{
+			OnTapped(sender);
+		}
+	}
+
+	internal void HandleUserInteraction(TouchBehavior sender, TouchInteractionStatus interactionStatus)
+	{
+		if (sender.InteractionStatus != interactionStatus)
+		{
+			sender.InteractionStatus = interactionStatus;
+			sender.RaiseInteractionStatusChanged();
+		}
+	}
+
+	internal void HandleHover(TouchBehavior sender, HoverStatus status)
+	{
+		if (!sender.Element?.IsEnabled ?? true)
+		{
+			return;
+		}
+
+		var hoverState = status == HoverStatus.Entered
+			? HoverState.Hovered
+			: HoverState.Normal;
+
+		if (sender.HoverState != hoverState)
+		{
+			sender.HoverState = hoverState;
+			sender.RaiseHoverStateChanged();
+		}
+
+		if (sender.HoverStatus != status)
+		{
+			sender.HoverStatus = status;
+			sender.RaiseHoverStatusChanged();
+		}
+	}
+
+	internal async Task ChangeStateAsync(TouchBehavior sender, bool animated)
+	{
+		var status = sender.Status;
+		var state = sender.State;
+		var hoverState = sender.HoverState;
+
+		AbortAnimations(sender);
+		animationTokenSource = new CancellationTokenSource();
+		var token = animationTokenSource.Token;
+
+		var isToggled = sender.IsToggled;
+
+		if (sender.Element is not null)
+		{
+			UpdateVisualState(sender.Element, state, hoverState);
+		}
+
+		if (!animated)
+		{
+			if (isToggled.HasValue)
+			{
+				state = isToggled.Value
+					? TouchState.Pressed
+					: TouchState.Normal;
+			}
+
+			var durationMultiplier = this.durationMultiplier;
+			this.durationMultiplier = null;
+
+			await RunAnimationTask(sender, state, hoverState, animationTokenSource.Token, durationMultiplier.GetValueOrDefault()).ConfigureAwait(false);
+			return;
+		}
+
+		var pulseCount = sender.PulseCount;
+
+		if (pulseCount == 0 || (state == TouchState.Normal && !isToggled.HasValue))
+		{
+			if (isToggled.HasValue)
+			{
+				Console.WriteLine($"Touch state: {status}");
```

Comment:
Please, change all `Console.WriteLine` to `Trace.WriteLine` the reason is `Console.*` is trimmed on Release builds, so users will not have a nice log.

Reply:
Updated